spec/mongoid/config/introspection_spec.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe Mongoid::Config::Introspection do
+
+  context "CONFIG_RB_PATH" do
+    it "refers to an actual source file" do
+      expect(File.exist?(Mongoid::Config::Introspection::CONFIG_RB_PATH)).to be true
+    end
+  end
+
+  context "#options" do
+    let(:options) { Mongoid::Config::Introspection.options }
+    let(:all_options) { Mongoid::Config::Introspection.options(include_deprecated: true) }
+    let(:deprecated_options) { all_options.select(&:deprecated?) }
+
+    # NOTE: the "with deprecated options" context tests a configuration option
+    # that is known to be deprecated (see `let(:option_name)`). When this
+    # deprecated option is eventually removed, the "exclude" spec will break.
+    # At that time, update the `:option_name` helper to reference a different
+    # deprecated option (if any), or else skip the specs in this context.
+    context "with deprecated options" do
+      let(:option_name) { "background_indexing" }
+
+      it "should exclude deprecated options by default" do
+        option = options.detect { |opt| opt.name == option_name }
+        expect(option).to be_nil
+      end
+
+      it "deprecated options should be included when requested" do
+        option = all_options.detect { |opt| opt.name == option_name }
+        expect(option).not_to be_nil
+      end
+    end
+
+    Mongoid::Config.defaults.each do |name, default_value|
+      context "for the `#{name}` option" do
+        let(:option) { all_options.detect { |opt| opt.name == name.to_s } }
+        let(:live_option) { options.detect { |opt| opt.name == name.to_s } }
+
+        it "should be parsed by the introspection scraper" do
+          expect(option).not_to be_nil
+          expect(eval(option.default)).to be == default_value
```

Comment:
`eval` even in tests should be avoided. Is there anyway this could be done with a regexp?

Reply:
Yeah, that's fair. I think I can turn it around and compare `option.default` to to `default_value.inspect`. I'll experiment.